nexus/src/app/sagas/instance_create.rs

```diff
@@ -615,45 +707,47 @@ async fn sic_create_disks_for_instance(
 /// Undo disks created during instance creation
 // TODO implement
 async fn sic_create_disks_for_instance_undo(
-    _sagactx: ActionContext<SagaInstanceCreate>,
-    _disk_index: usize,
+    _sagactx: NexusActionContext,
 ) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
 async fn sic_attach_disks_to_instance(
-    sagactx: ActionContext<SagaInstanceCreate>,
-    disk_index: usize,
+    sagactx: NexusActionContext,
 ) -> Result<(), ActionError> {
-    ensure_instance_disk_attach_state(sagactx, disk_index, true).await
+    ensure_instance_disk_attach_state(sagactx, true).await
 }
 
 async fn sic_attach_disks_to_instance_undo(
-    sagactx: ActionContext<SagaInstanceCreate>,
-    disk_index: usize,
+    sagactx: NexusActionContext,
 ) -> Result<(), anyhow::Error> {
-    Ok(ensure_instance_disk_attach_state(sagactx, disk_index, false).await?)
+    Ok(ensure_instance_disk_attach_state(sagactx, false).await?)
 }
 
 async fn ensure_instance_disk_attach_state(
-    sagactx: ActionContext<SagaInstanceCreate>,
-    disk_index: usize,
+    sagactx: NexusActionContext,
     attached: bool,
 ) -> Result<(), ActionError> {
     let osagactx = sagactx.user_data();
-    let saga_params = sagactx.saga_params();
+    let disk_params = sagactx.saga_params::<DiskParams>()?;
+    let saga_params = disk_params.saga_params;
+    let disk_index = disk_params.which;
     let opctx =
         OpContext::for_saga_action(&sagactx, &saga_params.serialized_authn);
 
     let saga_disks = &saga_params.create_params.disks;
-    let instance_name = sagactx.lookup::<db::model::Name>("instance_name")?;
+    let instance_name =
+        db::model::Name(saga_params.create_params.identity.name);
 
     if disk_index >= saga_disks.len() {
         return Ok(());
     }
 
     let disk = &saga_disks[disk_index];
 
+    // TODO-correctness TODO-security It's not correct to re-resolve the
```

Comment:
This seems important: Maybe open an issue for it?

Reply:
Filed https://github.com/oxidecomputer/omicron/issues/1536 and updated this comment to reference it.